### PR TITLE
feat(flightsql): Support `GetTableTypes` metadata API

### PR DESCRIPTION
**Pull request by alamb**
Part of https://github.com/influxdata/influxdb_iox/issues/7210

This PR implement the `GetTableTypes` FlightSQL metadata endpoint, with tests.

It is implementing using the same SQL as is used by `queryrouterd`:

```sql
SELECT DISTINCT table_type FROM information_schema.tables ORDER BY table_type
```

https://github.com/influxdata/idpe/blob/85aa7a52b40f173cc4d79ac02b3a4a13e82333c4/queryrouter/internal/server/flightsql_handler.go#L299-L318

### CODE REVIEW

File: influxdb_iox/tests/jdbc_client/Main.java [hunk 1]
```diff
@@ -132,6 +132,11 @@ static void run_metadata(String url) throws SQLException {
         System.out.println("**************");
         print_result_set(md.getSchemas());
 
+        System.out.println("**************");
+        System.out.println("Table Types:");
+        System.out.println("**************");
+        print_result_set(md.getTableTypes());
```
Comment (alamb) [inline influxdb_iox/tests/jdbc_client/Main.java:138]: here is the code that calls the API via JDBC (this is verified in the `influxdb_iox/tests/end_to_end_cases/flightsql.rs`)

**Review by crepererum — APPROVED** [review]
awesome progress :muscle:

**Review by appletreeisyellow — APPROVED** [review]
This is a good one for me to start with on FlightSQL metadata! I left two questions for my understanding

**Review by mhilton — APPROVED** [review]

File: flightsql/src/planner.rs [hunk 2]
```diff
@@ -54,6 +54,10 @@ impl FlightSQLPlanner {
                 let plan = plan_get_db_schemas(ctx, catalog, db_schema_filter_pattern).await?;
                 get_schema_for_plan(plan)
             }
+            FlightSQLCommand::CommandGetTableTypes(CommandGetTableTypes {}) => {
+                let plan = plan_get_table_types(ctx).await?;
+                get_schema_for_plan(plan)
```
[the remark this answers is not in the document]
Comment (alamb) [reply flightsql/src/planner.rs:65]: Tracking in https://github.com/influxdata/influxdb_iox/issues/7278

[the remark this answers is not in the document]
Comment (alamb) [reply flightsql/src/planner.rs:65]: PR to fix: https://github.com/influxdata/influxdb_iox/pull/7281

**Pull request merged**